### PR TITLE
fix(@clayui/form): ClayToggle should use a label element instead of span

### PR DESCRIPTION
**Pull request by bryceosterhaus**
fixes https://github.com/liferay/clay/issues/3838

### CODE REVIEW

**Review by markocikos — APPROVED** [review]
LGTM!

**Review by matuzalemsteles — COMMENTED** [review]
Hey @bryceosterhaus looking at the markup it seems that we no longer have the `.toggle-switch-check-bar` wrapper, @pat270 can you confirm that this is it?

File: packages/clay-form/src/Toggle.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ const ClayToggle = React.forwardRef<HTMLLabelElement, IToggleProps>(
 		}
 
 		return (
-			<span
+			<label
 				className={classNames(
 					'toggle-switch',
 					'simple-toggle-switch',
```
Comment (pat270) [inline packages/clay-form/src/Toggle.tsx:51]: Some users might not want to use `simple-toggle-switch`. I've seen commerce use toggle switches in narrow columns that make `simple-toggle-switch` look funny. Do we have a way to only use `toggle-switch`?

**Comment by pat270** [comment]
@matuzalemsteles `.toggle-switch-check-bar` is new from v2.11.1+. It aligns the hidden checkbox to the toggle-switch overlay. Our documentation on clayui.com is incorrect.

**Pull request merged**